public/app/features/explore/QueryRow.tsx

```diff
@@ -136,44 +137,52 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
       QueryField = datasourceInstance.components?.ExploreQueryField;
     }
 
+    const filteredQueryResponse: PanelData = {
+      ...queryResponse,
+      error: null,
+    };
+
     return (
-      <div className="query-row">
-        <div className="query-row-field flex-shrink-1">
-          {QueryField ? (
-            <QueryField
-              datasource={datasourceInstance}
-              query={query}
-              history={history}
-              onRunQuery={this.onRunQuery}
-              onBlur={noopOnBlur}
-              onChange={this.onChange}
-              data={queryResponse}
-              absoluteRange={absoluteRange}
-              exploreMode={mode}
-            />
-          ) : (
-            <QueryEditor
-              error={queryErrors}
-              datasource={datasourceInstance}
-              onQueryChange={this.onChange}
-              onExecuteQuery={this.onRunQuery}
-              initialQuery={query}
-              exploreEvents={exploreEvents}
-              range={range}
-              textEditModeEnabled={this.state.textEditModeEnabled}
-            />
-          )}
+      <>
+        <div className="query-row">
+          <div className="query-row-field flex-shrink-1">
+            {QueryField ? (
+              <QueryField
+                datasource={datasourceInstance}
+                query={query}
+                history={history}
+                onRunQuery={this.onRunQuery}
+                onBlur={noopOnBlur}
+                onChange={this.onChange}
+                data={filteredQueryResponse}
+                absoluteRange={absoluteRange}
+                exploreMode={mode}
+              />
+            ) : (
+              <QueryEditor
+                error={[]}
```

Comment:
I am wondering whether we should be still passing the errors. On one hand there is a risk some editors will show the errors 2 times because of that, on the other hand we do not really know what the editor will do with the error, it may use it for some logic inside (maybe retry or disabling some buttons etc) so taking this info away seems like too much of a breaking change here.

@davkal do have any opinion on this?

Reply:
Good catch. I'll bring passing down errors back. We already took care of some of the plugins (Prometheus, Loki, Elastic) so they shouldn't display duplicated errors.